RecoEcal/EgammaCoreTools/src/Mustache.cc

```diff
@@ -33,114 +23,67 @@ namespace reco {
       //b comes from a fit to the width
       //and has a slight dependence on E on the upper edge
       // this only works because of fine tuning :-D
-      const float sqrt_log10_clustE = std::sqrt(std::log10(ClustE) + 1.1);
-      // we need to have this in two steps, so that we don't improperly shift
-      // the lower bound!
-      float b_upper = w10 * eta0xsineta0 + w11 / sqrt_log10_clustE;
-      float b_lower = w00 * eta0xsineta0 + w01 / sqrt_log10_clustE;
-      const float midpoint = 0.5 * (b_upper + b_lower);
-      b_upper -= midpoint;
-      b_lower -= midpoint;
+      const float sqrt_log10_clustE = std::sqrt(log10ClustE + params->sqrtLogClustETuning());
+      const float b_upper =
+          parabola_params.w1Up[0] * eta0xsineta0 + parabola_params.w1Up[1] / sqrt_log10_clustE -
+          0.5 * (parabola_params.w1Up[0] * eta0xsineta0 + parabola_params.w1Up[1] / sqrt_log10_clustE +
+                 parabola_params.w0Up[0] * eta0xsineta0 + parabola_params.w0Up[1] / sqrt_log10_clustE);
+      const float b_lower =
+          parabola_params.w0Low[0] * eta0xsineta0 + parabola_params.w0Low[1] / sqrt_log10_clustE -
+          0.5 * (parabola_params.w1Low[0] * eta0xsineta0 + parabola_params.w1Low[1] / sqrt_log10_clustE +
+                 parabola_params.w0Low[0] * eta0xsineta0 + parabola_params.w0Low[1] / sqrt_log10_clustE);
 
       //the curvature comes from a parabolic
       //fit for many slices in eta given a
       //slice -0.1 < log10(Et) < 0.1
-      const float curv_up = eta0xsineta0 * (p00 * eta0xsineta0 + p01) + p02;
-      const float curv_low = eta0xsineta0 * (p10 * eta0xsineta0 + p11) + p12;
+      const float curv_up =
+          eta0xsineta0 * (parabola_params.pUp[0] * eta0xsineta0 + parabola_params.pUp[1]) + parabola_params.pUp[2];
+      const float curv_low =
+          eta0xsineta0 * (parabola_params.pLow[0] * eta0xsineta0 + parabola_params.pLow[1]) + parabola_params.pLow[2];
 
       //solving for the curviness given the width of this particular point
-      const float a_upper = (1 / (4 * curv_up)) - fabs(b_upper);
-      const float a_lower = (1 / (4 * curv_low)) - fabs(b_lower);
+      const float a_upper = (1 / (4 * curv_up)) - std::abs(b_upper);
+      const float a_lower = (1 / (4 * curv_low)) - std::abs(b_lower);
 
       const double dphi = TVector2::Phi_mpi_pi(ClusPhi - maxPhi);
       const double dphi2 = dphi * dphi;
       // minimum offset is half a crystal width in either direction
       // because science.
-      const float upper_cut = (std::max((1. / (4. * a_upper)), 0.0) * dphi2 + std::max(b_upper, 0.0087f)) + 0.0087;
-      const float lower_cut = (std::max((1. / (4. * a_lower)), 0.0) * dphi2 + std::min(b_lower, -0.0087f));
+      constexpr float half_crystal_width = 0.0087;
```

Comment:
In addition, I think a comment or reference to the origin would be needed here. Is it the same parameter as https://github.com/cms-sw/cmssw/blob/9c9e72f134d6727f96fbbed3a538b4c408e2e8d6/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc#L79

Reply:
To be honest I do not know. It certainly looks very similar but I do not think the origin is in the L1Trigger package. The value comes from the detector geometry and I thought the variable name is descriptive enough.

Reply:
OK, that sounds reasonable to me. Anyway your implementation of making it a constexpr is already an improvement. Perhaps in the future this number might be accessed from geometry.

Reply:
OK I will leave it this way then.